Source/JavaScriptCore/heap/MarkedBlock.cpp

```diff
@@ -502,6 +511,88 @@ void MarkedBlock::Handle::sweep(FreeList* freeList)
     specializedSweep<false, IsEmpty, SweepOnly, BlockHasNoDestructors, DontScribble, HasNewlyAllocated, MarksStale>(freeList, emptyMode, sweepMode, BlockHasNoDestructors, scribbleMode, newlyAllocatedMode, marksMode, [] (VM&, JSCell*) { });
 }
 
+NO_RETURN_DUE_TO_CRASH NEVER_INLINE void MarkedBlock::dumpInfoAndCrashForInvalidHandle(AbstractLocker&, HeapCell* heapCell)
+{
+    JSCell* cell = bitwise_cast<JSCell*>(heapCell);
+    JSType cellType = cell->type();
+
+    StringPrintStream out;
+    out.printf("current markedblock = %p\n", this);
+    out.printf("found null handle in heap cell %p with type %d\n", heapCell, cellType);
+#if PLATFORM(COCOA)
+    WTF::setCrashLogMessage(out.toCString().data());
+#endif
```

Comment:
nit: Also, instead of printing 2 lines here, with this little info, there's no reason why this can't be a single line e.g.
```
    logInvalidHandleDetails("INVALID HANDLE: MarkedBlock %p HeapCell %p CellType %d\n", this, heapCell, cellType);
```
Note: I would prefix all these dumps with "INVALID HANDLE: ".  It gives us a handle to search for in the logs later (just in case it doesn't end up at the bottom for whatever reason).